vector/src/main/java/im/vector/util/MediaAVScanner.java

```diff
@@ -56,140 +40,128 @@ public MediaAVScanner(String userId) {
+                    scanStatus = mediaAntivirusScanStatus.getScanStatus();
+                }
+            } finally {
+                realm.close();
+            }
         }
+        return scanStatus;
 
-        mMediaAVScanByUrl = new HashMap<>();
-
-        try {
-            mMediaAVScannerDirectory = new File(context.getApplicationContext().getFilesDir(), MEDIAAVSCANNER_STORE_FOLDER);
-            mMediaAVScannerDirectory = new File(mMediaAVScannerDirectory, mUserId);
-
-            mMediaAVScannerFile = new File(mMediaAVScannerDirectory, FILENAME.hashCode() + "");
+    }
 
-            if (!mMediaAVScannerDirectory.exists()) {
+    /**
+     * Clear the Realm cached media scan results.
+     */
+    public void clearAntivirusScanResultsCache() {
 
-                // create dir tree
-                mMediaAVScannerDirectory.mkdirs();
+        realm.executeTransactionAsync(new Realm.Transaction() {
+            @Override
+            public void execute(Realm bgRealm) {
+                RealmResults<MediaAntivirusScanStatus> results = realm.where(MediaAntivirusScanStatus.class).findAll();
+                results.deleteAllFromRealm();
             }
-
-            if (mMediaAVScannerFile.exists()) {
-                FileInputStream fis = new FileInputStream(mMediaAVScannerFile);
-                ObjectInputStream ois = new ObjectInputStream(fis);
-                mMediaAVScanByUrl = (HashMap) ois.readObject();
-                ois.close();
-                fis.close();
+        }, new Realm.Transaction.OnSuccess() {
+            @Override
+            public void onSuccess() {
+                realm.close();
             }
-        } catch (Exception e) {
-            Log.e(LOG_TAG, "## openMediaAVScannerDict failed " + e.getMessage());
-        }
+        }, new Realm.Transaction.OnError() {
+            @Override
+            public void onError(Throwable error) {
+                Log.e(LOG_TAG, "## clearAntivirusScanResultsCache : Realm transaction failed");
+            }
+        });
     }
 
     /**
      * Get the current scan status for a media.
      * Trigger a scan if it is not already done.
      *
-     * @param context   the context.
      * @param url       the media url.
      * @param callback  optional async response handler.
      * @return the current scan status.
      */
-    public ScanStatus scanEncryptedMedia(Context context, String url, @Nullable ApiCallback<Void> callback) {
-        if (null == mMediaAVScanByUrl) {
-            openMediaAVScannerDict(context);
-        }
+    public MediaAntivirusScanStatus.ScanStatus scanMedia(String url, @Nullable ApiCallback<Void> callback) {
 
         if (TextUtils.isEmpty(url)) {
-            return ScanStatus.UNKNOWN;
+            return MediaAntivirusScanStatus.ScanStatus.UNKNOWN;
         }
 
-        if (mMediaAVScanByUrl.containsKey(url)) {
-            return mMediaAVScanByUrl.get(url);
+        getMediaAvScanByUrl(url);
```

Comment:
You are calling 3 times the same methods `getMediaAvScanByUrl(url)`. Please optimize :)